pkg/charts/utils.go

```diff
@@ -204,10 +215,57 @@ func ComputeCalicoChartValues(
 	return calicoChartValues, nil
 }
 
-func generateChartValues(config *calicov1alpha1.NetworkConfig, kubeProxyEnabled bool, nonPrivileged bool) (*calicoConfig, error) {
+func generateChartValues(network *extensionsv1alpha1.Network, config *calicov1alpha1.NetworkConfig, kubeProxyEnabled bool, nonPrivileged bool) (*calicoConfig, error) {
+	// by default assume IPv4 (dual-stack is not supported, yet)
+	ipFamilies := sets.New[extensionsv1alpha1.IPFamily](network.Spec.IPFamilies...)
+	isIPv4 := true
+	isIPv6 := false
+	if ipFamilies.Has(extensionsv1alpha1.IPFamilyIPv6) {
+		isIPv4 = false
+		isIPv6 = true
+	}
+
 	c := newCalicoConfig()
+	if isIPv4 {
+		c.IPAM.AssignIPv4 = true
+		c.IPAM.Subnet = usePodCIDR
+		c.IPv4 = ipv4{
+			Enabled:             true,
+			Pool:                calicov1alpha1.PoolIPIP,
+			Mode:                calicov1alpha1.Always,
+			AutoDetectionMethod: nil,
+		}
+	}
+
+	if isIPv6 {
+		c.IPAM.AssignIPv6 = true
+		c.IPAM.Subnet = usePodCIDRv6
+		// NOTE: calico guide (https://docs.tigera.io/calico/latest/networking/ipam/ipv6#enable-ipv6-only) requires ipam
+		// type to be calico-ipam for single IPv6 or dual-stack clusters.
+		c.IPAM.IPAMType = calicoIPAM
```

Comment:
Did we interpret this correctly? @DockToFuture

Reply:
CC: @axel7born

Reply:
Yes, unfortunately it's currently not possible to use the host-local IPAM plugin. I've raised an issue in regards to that: https://github.com/projectcalico/calico/issues/8190